### ISSUE TITLE
Description for user defined `cmd` options is not aligned with default options

### ISSUE THREAD

**Issue by MangelMaxime** [slaveOftime/Fun.Build#31]
When running `dotnet fsi build.fsx -- -p Repo --help` for the following pipeline

```fs
pipeline "Repo" {

    stage "Watch" {
        whenCmdArg "watch" "Watch the repo for changes"
        run "echo 0"
    }

    runIfOnlySpecified

}
```

we get this output:

![SCR-20230213-l03](https://user-images.githubusercontent.com/4760796/218479994-3e480021-2dca-400c-bbe3-6fc1612898a4.png)

The "problem" is that the description of `watch` is not aligned with the other descriptions.

**Comment by albertwoo** [comment]
@MangelMaxime hi, you should write like **whenCmdArg "watch" "" "Watch tests for changes"**. Because its defined as:
```fsharp
member inline _.whenCmdArg([<InlineIfLambda>] build: BuildStage, argKey: string, ?argValue: string, ?description)
```
The second value is argValue, not description.

**Comment by MangelMaxime** [comment]
Ah yes, I got bitten by this one again. I don't know if this is just me or because there is places for improvements on this subject.

In CLI it can happens that your CMD doesn't have an alias. Especially, if your "arg" is consider a command instead of an options.

- Command: `watch`
- Options: `--watch`

What is your opinion on providing?

```fs
member inline _.whenCmdArg([<InlineIfLambda>] build: BuildStage, argKey: string, ?description)
```

**Comment by albertwoo** [comment]
The problem is that, it is not alias, it is key value pair. Which means, when you say **whenCmdArg "key" "value"**, then in your command input, you will need to provide:
```bash
dotnet fsi xxx.fsx -- .... key value
```

So the printed info does not have a **,** but yeah, maybe I should print **key = value**

**Comment by MangelMaxime** [comment]
I was trying to propose a new spec to cover command, options, and key values but I am hitting a wall.

The problem, is that with the current types we can't make the distinction between:

- a command with a description
- a command with a value without providing an empty value

For Fun.Build, the following cases goes under a command with a value.

```fs
pipeline "Repo" {

    stage "Watch" {
        whenCmdArg "pull" "This is a description" 
        whenCmdArg "pull" "env"
        run "echo 0"
    }

    runIfOnlySpecified

}
```

*I think this is what you are trying to explain to me*

--------

Does that means when I do `whenCmdArg "-w" "--watch" "Watch tests for changes"`, I am actually expecting the user to provid:

`dotnet fsi build.fsx -- -w --watch` and not creating alias commands?

--------

Do you think the current situation is "problematic"?

Do you think we could use more specialised names?

- `whenCmdArg`
- `whenOption`
- `whenKeyValue`

Having a distinction between the `cmd` and `option` means we can generate a help with 2 distinct sections. (could already be done by testing if the provided value start with a `-` or not.

Do you think we could try to use better types/representation?

```fs
pipeline "Repo" {

    stage "Watch" {
		// For command

		// Describe: 
		// dotnet fsi build.fsx -- pull
		// dotnet fsi build.fsx -- pull something
        whenCmdArg {
			name "pull"
			description "This is a description"
		}

		// Describe: 
		// dotnet fsi build.fsx -- pull test
        whenCmdArg {
			name "pull"
			value "test"
		}

		// Describe: 
		// dotnet fsi build.fsx -- pull test
		// dotnet fsi build.fsx -- pull prod
        whenCmdArg {
			name "pull"
			acceptedValues [ "test"; "prod" ]
		}

		// For options

		// Describe: 
		// dotnet fsi build.fsx -- --env 
		// dotnet fsi build.fsx -- --env something
		whenOptions {
			prefix "env" // Correspond to `--env
			description "This is a description"
		}
		

		// Describe:
		// dotnet fsi build.fsx -- --env test
		whenOptions {
			prefix "env" // Correspond to `--env
			value "test"
		}		

        run "echo 0"
    }

    runIfOnlySpecified
}
```

The new style can be a replacement of the old API or leave in parallel, as they can be unified under the hood behind the same internal types.

**Comment by albertwoo** [comment]
> Does that means when I do whenCmdArg "-w" "--watch" "Watch tests for changes", I am actually expecting the user to provid: 
  dotnet fsi build.fsx -- -w --watch and not creating alias commands?

Yes, that is what mean.

The reason I called it whenCmdArg, is because **Environment.GetCommandLineArgs()**, cmd is indicating it is from command line. Like whenEnvArg, is from environment variables.

We should keep the **whenCmdArg**, because it is already used and to avoid conflict.

I like your proposal, what about we just create one more CE **whenCmd** and use it like:

```fsharp
whenCmd {
    name "-w"
    alias "--watch"
    description "some description ..."
    value "test"
    acceptedValues [ "test"; "prod" ]
}
```

**Comment by MangelMaxime** [comment]
I do like your **whenCmd** proposition.

**Comment by albertwoo** [comment]
Released in 0.3.3
